hooks/shotgun_get_published_file_for_viewer.py

```diff
@@ -44,13 +44,13 @@ def resolve_single_file(self, entity_type, published_file):
             raise TankError(
                 "Sorry, viewer extensions must be provided."
             )
-        path_on_disk = self.parent.get_path_on_disk(published_file)
+        path_on_disk = self.parent.get_path_on_disk(self.published_file(entity_type, published_file["id"]))
```

Comment:
This causes a SG query and we're calling it again below if we have a match. Maybe something like:

```python
sg_published_file = self.published_file(entity_type, published_file["id"])
path_on_disk = self.parent.get_path_on_disk(sg_published_file)
```
and then for the match:
```python
        ...
        if path_on_disk.endswith(".%s" % viewer_extension):
            return sg_published_file
```